.github/workflows/tests-nightly.yml

```
@@ -60,7 +60,8 @@ jobs:
 
       - name: Run tests
         run: |
-          pytest --ignore=tests/e2e/ tests/
+          pytest -n8 --dist loadfile --ignore=tests/e2e/ --ignore=tests/patched/ tests/
+          pytest tests/patched/
```

```suggestion
          pytest -n8 --ignore=tests/e2e/ --ignore=tests/patched/ tests/
          pytest -n1 --dist loadfile tests/patched/
```
I think this is inverted and we want loadfile on the patched tests

---

IIUC `-n1 --dist loadfile` is the same as passing no args (we're using one worker, so we're of course loading each file onto the same worker).

---

I can go either way on using `--dist loadfile` for the first run, the runtime might be better in either case (would have to test)/

---

yeah, I think you're right. i think the way we have it now is an evolution of starting with just `--dist loadfile` and eventually just moving it to one worker to prevent the wonky issues that happens when we monkey patch things and leaving the process in a patched state for a test that doesn't expect the patches.